modules/message-lane/src/instant_payments.rs

```diff
@@ -76,6 +76,7 @@ where
 
 /// Return account id of shared relayers-fund account that is storing all fees
 /// paid by submitters, until they're claimed by relayers.
-fn relayers_fund_account<AccountId: Default + Decode>() -> AccountId {
-	bridge_account_id(NO_INSTANCE_ID, MESSAGE_LANE_MODULE_PREFIX)
+fn relayers_fund_account<AccountId: Default + Encode + Decode>() -> AccountId {
+	let encoded_id = derive_account_id::<AccountId>(NO_INSTANCE_ID, SourceAccount::Account(AccountId::default()));
+	AccountId::decode(&mut &encoded_id[..]).unwrap_or_default()
```

Comment:
Same here, this is actually super dangerous, let's make the conversion explicit and let's force the user to provide one instead of relying on `Decoding` the data.

Reply:
Lol yeah I meant to make a note about this asking for alternatives to the `_or_default()` call 😅

Reply:
I changed this address to be a configuration option in the `message-lane` pallet instead of being derived on every call. The main implication (from what I can tell) is that now each instance of the pallet will need to choose a `RelayerFundAccount`, which may or may not be the same across instances. Before this change all instances used to use the same account.

I'm not sure if this has any serious implications, do you mind chiming in here @svyatonik? The commit where this change is introduced is here: 88e28a84a07634cd8cd0c45ccf57481aaa5b611b.